pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -235,6 +235,9 @@ type AlertmanagerSpec struct {
 	// If defined, it takes precedence over the `configSecret` field.
 	// This field may change in future releases.
 	AlertmanagerConfiguration *AlertmanagerConfiguration `json:"alertmanagerConfiguration,omitempty"`
+	// If you don't want the kubelet to automatically mount a ServiceAccount's API credentials, you can opt out of the default behavior.
+	// You can opt out of automounting API credentials on /var/run/secrets/kubernetes.io/serviceaccount/token for a pod by setting automountServiceAccountToken: false on the pod spec.
+	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty"`
```

Comment:
```suggestion
	// AutomountServiceAccountToken defines if the ServiceAccount token is mounted into Alertmanager's pod or not.
	// +optional
	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty"`
```

Reply:
I think that the original comment is useful because otherwise it might be unclear when this field is useful.

```suggestion
	// AutomountServiceAccountToken indicates whether a service account token should be automatically mounted in the pod.
	// If the service account has `automountServiceAccountToken: true`, set the field to `false` to opt out of automounting API credentials.
	AutomountServiceAccountToken *bool `json:"automountServiceAccountToken,omitempty"`
```

Reply:
Sounds good to me, let's not forget about the extra comment `// +optional` 🙂

Reply:
Sorry for the late response,
Added the latest comment from @simonpasquier, with the `// +optional`